### ISSUE TITLE
CMSIS RTOS API: Fix osMessagePut()/osMailPut() inconsecvent error when called from ISR

### ISSUE THREAD

**Issue by ilg-ul** [ARM-software/CMSIS#62]
Most other calls return `osErrorISR` when called from ISR.

`osMessagePut()` (and its sister `osMailPut()`), when called from ISR with a timeout different from 0, returns `osErrorParameter`, as if the problem would be the parameter, not the context where it is called from.

In [CMSIS++](http://micro-os-plus.github.io/cmsis-plus/) this problem is simply avoided by having distinct calls, with all blocking calls banned from ISR, and all `try_xxx()` calls allowed from ISRs.

**Comment by ReinhardKeil** [comment]
Related to POSIX error codes

**Comment by ilg-ul** [comment]
that's correct, once accepting the use of POSIX codes, the current inconsequent behaviour would no longer be an issue.
